hibernate-core/src/main/java/org/hibernate/cache/spi/QueryKey.java

```diff
@@ -85,7 +87,8 @@ public QueryKey(
 		this.firstRow = firstRow;
 		this.maxRows = maxRows;
 		this.tenantIdentifier = tenantIdentifier;
-		this.enabledFilterNames = enabledFilterNames;
+		//ensure enabledFilterNames is serializable
+		this.enabledFilterNames = Collections.unmodifiableSet(new HashSet<>(enabledFilterNames));
```

Comment:
Since we have to copy this anyway, I'd prefer to use a `String[]`. I see that it is hard to properly test this, so I won't mandate that you write that.

Reply:
A test could be as simple as running query with enabled filters that puts into the cache. Before closing the session, mutate the filters. Then in a new session, call the same query again with the same filters. This should lead to a cache miss with this bug, but should be a cache hit without it.

Reply:
I switched to String[] as requested.

I also added a test and it behaves as described, but I think it's a bit convoluted:
- I had to put it in the jcache module, because the default cache impl in tests does not fail on non-serializable keys (I did not check, but it's probably just a Map)
- In the jcache module, EHCache won't throw an error when encountering a non-serializable key, because of the default [resilience strategy](https://www.ehcache.org/documentation/3.7/resilience.html), but won't cache the value either
- I'm using Caffeine as the JSR107 implementationin my project, which fails with a serialization exception instead of failing silently without actually caching, and made the bug much more obvious
=> Isn't it possible to configure EHCache (at least for the hibernate tests) to a resilience strategy that would throw an exception (ThrowingResilienceStrategy seems to do exaclty that)

I also changed the base branch to 6.2, let me know if I need to rebase on main.